internal/versions/versions.go

```diff
@@ -91,13 +94,24 @@ func (h *handler) V2ListComponentVersions(ctx context.Context, params operations
 
 func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params operations.V2ListSupportedOpenshiftVersionsParams) middleware.Responder {
 	openshiftVersions := models.OpenshiftVersions{}
+	hasArmAuthorization := false
+
+	err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
+	if err == nil {
+		hasArmAuthorization = true
+	}
+
 	for _, releaseImage := range h.releaseImages {
 		key := *releaseImage.OpenshiftVersion
 		if swag.StringValue(releaseImage.CPUArchitecture) == "" {
 			// Empty implies default architecture
 			*releaseImage.CPUArchitecture = common.DefaultCPUArchitecture
 		}
 
+		if !hasArmAuthorization && swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
```

Comment:
IMO (also, this is what I did in my PR), you should call `auth.ValidateAccessToCPUArchitecture` only when `swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture`.

That way, we would check for the capability only if it is relevant, saving an API call to AMS.

Reply:
But why call `auth.ValidateAccessToCPUArchitecture` whenever  `swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture` ? 
In this implementation we call it once at the beginning of the function

Reply:
But we can do both

```go
func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params operations.V2ListSupportedOpenshiftVersionsParams) middleware.Responder {
	openshiftVersions := models.OpenshiftVersions{}
	hasArmAuthorization := false

	for _, releaseImage := range h.releaseImages {
		key := *releaseImage.OpenshiftVersion
		if swag.StringValue(releaseImage.CPUArchitecture) == "" {
			// Empty implies default architecture
			*releaseImage.CPUArchitecture = common.DefaultCPUArchitecture
		}

		if !hasArmAuthorization && swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
			err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
			if err == nil {
				hasArmAuthorization = true
			}
			continue
		}
```